src/core/thread/discover_scanner.cpp

```diff
@@ -144,6 +157,24 @@ otError DiscoverScanner::Discover(const Mac::ChannelMask &aScanChannels,
     return error;
 }
 
+otError DiscoverScanner::SetJoinerAdvertisement(uint32_t aOui, const uint8_t *aAdvData, uint8_t aAdvDataLength)
+{
+    otError error = OT_ERROR_NONE;
+
+    VerifyOrExit((aAdvData != nullptr) && (aAdvDataLength != 0) && (aAdvDataLength <= kMaxLength),
+                 error = OT_ERROR_INVALID_ARGS);
+
+    mOui           = aOui;
+    mAdvDataLength = aAdvDataLength;
+
+    memcpy(mAdvData, aAdvData, aAdvDataLength);
```

Comment:
```suggestion
if (aAdvDataLength > 0)
{
    memcpy(mAdvData, aAdvData, aAdvDataLength);
}
```
Passing `nullptr` to memcpy is undefined?

Reply:
- maybe I miss something, but what's the intention to check if an `uint8_t` variable is greater than 0 or not?
- `nullptr` has been disallowed and won't be passed to `memcpy`, do you mean `nullptr` should be supported?

Reply:
Didn't notice that `nullptr` is prohibited.